tests/unit/metrics/test_ffwd.py

```diff
@@ -0,0 +1,208 @@
+    ('other-name', 'other-metric', 6.243289, {'hostname': 'dns-host-3'}, None),
+    ('service-name', 'other-metric', 500000, None, ['v40']),
+    ('service-name', 'other-metric', 500000, {'type': 'active'}, ['v40'])
+])
+def test_ffwdrelay_create_metric(test_config, key, name, val, attrs, tags):
+    """Creates metric for all arguments."""
+    test_config['key'] = key
+    relay = ffwd.FfwdRelay(test_config)
+    actual = relay._create_metric(name, val, attrs=attrs, tags=tags)
+
+    attrs = attrs or {}
+    attrs.update({'what': name})
+    expected = {
+        'key': key,
+        'attributes': attrs,
+        'value': val,
+        'type': 'metric',
+        'tags': tags or []
+    }
+    assert expected == actual
+
+
+@pytest.fixture
+def ffwdrelay_and_metrics_sent(test_config, mock_udp_client_and_metrics_sent):
+    mock_udp_client, metrics_sent = mock_udp_client_and_metrics_sent
+    relay = ffwd.FfwdRelay(test_config)
+    relay.udp_client = mock_udp_client
+    return relay, metrics_sent
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('attrs,tags', [
+    (None, None),
+    ({'hostname': 'dns-host-3'}, None),
+    (None, ['v40']),
+    ({'type': 'active'}, ['v40'])
+])
+async def test_ffwdrelay_incr(ffwdrelay_and_metrics_sent, attrs, tags):
+    """Initializes and sends metrics."""
+    ffwdrelay, metrics_sent = ffwdrelay_and_metrics_sent
+    expected = [
+        ffwdrelay._create_metric('some-counter', 1, attrs=attrs, tags=tags)
+    ]
+    await ffwdrelay.incr('some-counter', attrs=attrs, tags=tags)
+    assert expected == metrics_sent
+
+
+@pytest.fixture
+def ffwdrelay(ffwdrelay_and_metrics_sent):
+    return ffwdrelay_and_metrics_sent[0]
+
+
+@pytest.mark.parametrize('attrs,tags', [
+    (None, None),
+    ({'hostname': 'dns-host-3'}, None),
+    (None, ['v40']),
+    ({'type': 'active'}, ['v40'])
+])
+def test_ffwdrelay_timer(ffwdrelay, attrs, tags):
+    """Returns initialized timer."""
+    expected_metric = ffwdrelay._create_metric(
+        'some-timer', None, attrs=attrs, tags=tags)
+    actual = ffwdrelay.timer('some-timer', attrs=attrs, tags=tags)
+    assert isinstance(actual, ffwd.FfwdTimer)
+    assert expected_metric == actual.metric
+    assert ffwdrelay.time_unit == actual.time_unit
+    assert ffwdrelay.udp_client == actual.udp_client
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('attrs,tags', [
+    (None, None),
+    ({'hostname': 'dns-host-3'}, None),
+    (None, ['v40']),
+    ({'type': 'active'}, ['v40'])
+])
+async def test_ffwdrelay_set(ffwdrelay_and_metrics_sent, attrs, tags):
+    """Sends metric."""
+    ffwdrelay, metrics_sent = ffwdrelay_and_metrics_sent
+    expected = [
+        ffwdrelay._create_metric('some-counter', 42, attrs=attrs, tags=tags)
+    ]
+    await ffwdrelay.set('some-counter', 42, attrs=attrs, tags=tags)
+    assert expected == metrics_sent
+
+
+@pytest.mark.asyncio
+async def test_cleanup(ffwdrelay):
+    """Raises no errors."""
+    await ffwdrelay.cleanup()
```

Comment:
Technically the way it's written now, won't make tests fail since they'll just error out, but the build will go red, which I generally think is good enough.  @econchick any thoughts?

Reply:
Yeah I'm cool with how it is now.